src/apps/cardano/sign_tx.py

```diff
@@ -73,13 +73,30 @@ def is_change(output, inputs):
     progress.init(msg.transactions_count, "Loading data")
 
     try:
+        attested = len(msg.inputs)*[False]
+        input_coins = 0
         # request transactions
-        transactions = []
         tx_req = CardanoTxRequest()
+
         for index in range(msg.transactions_count):
             progress.advance()
             tx_ack = await request_transaction(ctx, tx_req, index)
-            transactions.append(tx_ack.transaction)
+            tx_hash = hashlib.blake2b(data=bytes(tx_ack.transaction), outlen=32).digest()
+            tx_decoded = cbor.decode(tx_ack.transaction)
+            for i, input in enumerate(msg.inputs):
+                if bytes(input.prev_hash) == tx_hash:
+                   attested[i] = True
+                   outputs = tx_decoded[1]
+                   amount = outputs[input.prev_index][1]
+                   input_coins += amount
+
+        for index in range(len(msg.inputs)):
```

Comment:
This could be simplified like this:
```
if (not all(attested)):
    raise wire.ProcessError("No tx data sent for input " + attested.index(False))
```